### PR TITLE
Include version info on header banner

### LINKED ISSUE #803: Include version info on header banner
**What would you like to be added**:

Let's find a way to include the current git SHA for the commit we've built and shipped on staging/production.  What if we altered this slightly so it was:

```html
<a href="https://github.com/Seneca-CDOT/telescope/commit/9e50064a8a090fd93975e8f037fe67ef5688fa0c" title="git commit 9e50064a8a090fd93975e8f037fe67ef5688fa0c">v0.7.0</a>
```

This would require the following changes:

- Instead of pulling in the entire `package.json` file [here](https://github.com/Seneca-CDOT/telescope/blob/master/src/frontend/src/components/Banner/Banner.jsx#L4), let's do it in https://github.com/Seneca-CDOT/telescope/blob/master/src/frontend/gatsby-config.js at build-time, and add the version and current git sha so we can build our URL for that link.  We can use something like https://www.npmjs.com/package/git-repo-info to pull this info into the gastby-config.js file.
- Add the version and git sha properties to our sitemetadata hook https://github.com/Seneca-CDOT/telescope/blob/master/src/frontend/src/hooks/use-site-metadata.js and StaticQuery https://github.com/Seneca-CDOT/telescope/blob/master/src/frontend/src/pages/index.js
- Do some CSS to keep the colour/text-decoration like it is now on the banner, but when you hover we can add `text-decoration: underline`.

**Why would you like this to be added**:

We currently do this, and on staging it doesn't indicate exactly which commit we are looking at:

Screen Shot 2020-03-05 at 12 27 00 PM

### PR DESCRIPTION
**Pull request by c3ho**
## Issue This PR Addresses
Fixes #803 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Version + SHA should now show up on banner.
Installed a package and moved all the info pulling into gatsby.config site metadata part. 

image

ZEIT is failing I think because I installed git-repo-info into our main package.json instead of the front-end one (I tried installing it here and pulling data from here, it didn't work). To use the module from there, I'm pulling it from node_modules.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by cindyledev** [comment]
For the underline on hover, you can add this to the `version` style in the `Banner.jsx`: 
```
    textDecorationLine: 'none',
    '&:hover': {
      textDecorationLine: 'underline',
    },
```

File: src/frontend/gatsby-config.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
 const path = require('path');
 
+const Version = require('../../package.json');
+
+const getRepoInfo = require('../../node_modules/git-repo-info');
```
Comment (humphd) [inline src/frontend/gatsby-config.js]: You should be able to `require('git-repo-info')` and the way that node resolves is that it will go up to parent directories to find it.  That said, why not just install this in the frontend package.json vs. the root?

File: src/frontend/gatsby-config.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+
+const info = getRepoInfo();
```
Comment (humphd) [inline src/frontend/gatsby-config.js]:
```js
const { sha } = getRepoInfo();
```

**Comment by c3ho** [comment]
Alright, I goofed. I found the issue, was trying to do `const getRepoInfo = require('./package.json')` when I was supposed to do `const getRepoInfo = require('git-repo-info')`

File: src/frontend/gatsby-config.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 const path = require('path');
+const getRepoInfo = require('git-repo-info');
```
Comment (humphd) [inline src/frontend/gatsby-config.js]:
```js
const { sha } = require('git-repo-info')();
```

File: src/frontend/gatsby-config.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
...
+const Version = require('../../package.json');
```
Comment (humphd) [inline src/frontend/gatsby-config.js]:
```js
const { version } = require('../../package.json');`
```

File: src/frontend/gatsby-config.js (hunk 3 above)
Comment (c3ho) [reply src/frontend/gatsby-config.js]: ah so much to learn

File: src/frontend/gatsby-config.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
 const path = require('path');
+const { sha } = require('git-repo-info')();
```
Comment (humphd) [inline src/frontend/gatsby-config.js]: If we can't use this, let's revert to what you did before (even though zeit/gatsby/webpack are wrong!)

Comment (c3ho) [reply src/frontend/gatsby-config.js]: Zeit just hasn't liked this pr very much since the start, the original way nor the way you suggested. Everything that has worked locally fails to build on zeit. Do we just abandon this pr?

**Comment by c3ho** [comment]
This is going to be stalled until... I have no idea when

**Comment by cindyledev** [comment]
This is so sad... we tried idk how many different methods and Zeit has rejected them all...
Annotation 2020-03-26 083233

Annotation 2020-03-26 083233_

**Comment by cindyledev** [comment]
Wait what does this mean? https://zeit.co/docs/v2/git-integrations/zeit-now-for-github
Included Environment Variables `NOW_GITHUB_COMMIT_SHA`. We had access to this the whole time?

**Comment by humphd** [comment]
@cindyledev we'd be able to use `NOW_GITHUB_COMMIT_SHA` on Zeit, but not with local dev or on staging, where it wouldn't exist.  So we'd have a new set of failures on everything but Zeit.

Good catch though.  I didn't know they had this.

**Comment by humphd** [comment]
See #882.  It would let you use a `fetch()` to get this infro from `/health`

File: package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@seneca/telescope",
   "private": true,
-  "version": "0.8.2",
+  "version": "0.0.1",
```
Comment (humphd) [inline package.json]: Let's not do this.  Can you roll back this commit?

Comment (c3ho) [reply package.json]: Sorry I checked out the wrong package.json should be fixed now

File: src/frontend/gatsby-config.js [hunk 7]
```diff
@@ -24,6 +25,7 @@ module.exports = {
     description: `A tool for tracking blogs in orbit around Seneca's open source involvement`,
     author: `SDDS Students and professors`,
     telescopeUrl,
+    version,
```
Comment (humphd) [inline src/frontend/gatsby-config.js]: We probably don't need the changes to this file if we're going to get it via /health.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 8]
```diff
@@ -146,7 +149,7 @@ function RetrieveBannerDynamicAssets() {
   useEffect(() => {
     async function getBackgroundImgSrc() {
       // Uses https://unsplash.com/collections/894/earth-%26-planets collection
-      /* Other Options: 
+      /* Other Options:
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]: Careful with unrelated changes like this.  It's fine here, but don't change git blame if you don't have to: it makes it harder to understand the history of this file.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 9]
```diff
@@ -218,6 +221,24 @@ ScrollDown.propTypes = {
...
 export default function Banner() {
   const classes = useStyles();
+  const { version, telescopeUrl } = useSiteMetadata();
+  const [sha, setSha] = useState('');
+
+  async function getSha() {
+    try {
+      const res = await fetch(`${telescopeUrl}/health`);
+      const data = await res.json();
+      return data.info.sha;
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]: Pull version out of here too.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 10]
```diff
@@ -218,6 +221,24 @@ ScrollDown.propTypes = {
 
 export default function Banner() {
   const classes = useStyles();
+  const { version, telescopeUrl } = useSiteMetadata();
+  const [sha, setSha] = useState('');
+
+  async function getSha() {
+    try {
+      const res = await fetch(`${telescopeUrl}/health`);
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]: Error handling please.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 11]
```diff
@@ -218,6 +221,24 @@ ScrollDown.propTypes = {
...
+      return data.info.sha;
+    } catch (error) {
+      console.error(`Error retrieving site's health info`, error);
+      return error;
+    }
+  }
+
+  useEffect(async () => {
+    setSha(await getSha());
+  }, []);
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]:
```js
}, [telescopeUrl]);
```

File: src/frontend/src/components/Banner/Banner.jsx [hunk 12]
```diff
@@ -230,8 +251,13 @@ export default function Banner() {
           </Typography>
         </ThemeProvider>
 
-        <div className={classes.version}>v {Version.version}</div>
-
+        <a
+          href={`https://github.com/Seneca-CDOT/telescope/commit/${sha}`}
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]: I think `/health` has a full GitHubUrl already built that you can use.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 13]
```diff
@@ -221,23 +221,35 @@ ScrollDown.propTypes = {
...
-      const data = await res.json();
-      return data.info.sha;
+
+      // Fetch failure
+      if (!res.ok) {
+        throw new Error(res.statusText);
+      }
+
+      // Checking whether content type is correct
+      if (res.headers.get('content-type').includes('application/json')) {
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]: Don't bother with this.  Ray did it because he was re-using the function, but we know it will be JSON.

Comment (c3ho) [reply src/frontend/src/components/Banner/Banner.jsx]: Removed.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 14]
```diff
@@ -218,6 +221,34 @@ ScrollDown.propTypes = {
 
 export default function Banner() {
   const classes = useStyles();
+  const { telescopeUrl } = useSiteMetadata();
+  const [sha, setSha] = useState('');
+  const [version, setVersion] = useState('');
+  const [githubUrl, setGitHubUrl] = useState('');
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]: Don't bother with this now, but you could use a single state Object vs. splitting this out into three.  It's fine for state to be a nested object in react.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 15]
```diff
@@ -218,6 +221,34 @@ ScrollDown.propTypes = {
...
+        throw new Error(res.statusText);
+      }
+
+      const data = await res.json();
+      setSha(data.info.sha);
+      setGitHubUrl(data.info.gitHubUrl);
+      setVersion(data.info.version);
+    } catch (error) {
+      console.error(`Error retrieving site's health info`, error);
+      throw new Error('Error retrieving site health info', error);
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]: Who are you throwing this to?  Drop this line.

File: src/frontend/src/components/Banner/Banner.jsx [hunk 16]
```diff
@@ -218,6 +221,34 @@ ScrollDown.propTypes = {
...
+    }
+  }
+
+  useEffect(async () => {
+    await getGitData();
```
Comment (humphd) [inline src/frontend/src/components/Banner/Banner.jsx]: You're not allowed to do this in a hook (they have to return `undefined` vs. a `Promise`).  Change it like this:

```js
useEffect(() => {
  async function getGitData() {...}

  getGitData();
});
```

File: src/frontend/src/components/Banner/Banner.jsx [hunk 17]
```diff
@@ -91,6 +89,11 @@ const useStyles = makeStyles(theme => ({
       left: '80%',
       bottom: theme.spacing(18),
     },
+    color: 'white',
+    textDecorationLine: 'none',
+    '&:hover': {
+      textDecorationLine: 'underline',
+    },
```
Comment (cindyledev) [inline src/frontend/src/components/Banner/Banner.jsx]: This is cute, you put this in `icon` instead of `version` XD

Comment (cindyledev) [inline src/frontend/src/components/Banner/Banner.jsx]: This is cute, you put this in `icon` instead of `version` XD

Comment (c3ho) [reply src/frontend/src/components/Banner/Banner.jsx]: The rebase was going weird on me
